speaker/bgp_controller.go

```diff
@@ -210,7 +210,10 @@ func (c *bgpController) syncPeers(l log.Logger) error {
 func (c *bgpController) SetBalancer(l log.Logger, name string, lbIP net.IP, pool *config.Pool) error {
 	c.svcAds[name] = nil
 	for _, adCfg := range pool.BGPAdvertisements {
-		m := net.CIDRMask(adCfg.AggregationLength, 32)
+		m := net.CIDRMask(128, 128)
+		if len(lbIP) == net.IPv4len {
```

Comment:
same here, you need to check if ip.To4() != nil if you want to check for IPv4-ness.